### PR TITLE
[Zizheng] iP

### PR DESCRIPTION
**Pull request by dr-arrgghh**

### CODE REVIEW

File: src/main/java/Duke.java [hunk 1]
```diff
@@ -1,10 +1,143 @@
...
+    public void chatDuke() throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        ArrayList<Task> taskArray = new ArrayList<>();
+        String str = sc.nextLine();
+        int counter = 0;
+        if (str.equals("bye")) {
+            System.out.println("Bye. Hope to see you again soon");
+        }
+        while (!str.equals("bye")){
+            if (str.equals("list")) {
```
Comment (Prearo) [inline src/main/java/Duke.java]: Consider using a switch statement here to reduce clutter and reduce nesting of if-else clauses.

File: src/main/java/Duke.java [hunk 2]
```diff
@@ -1,10 +1,143 @@
+import java.util.*;
```
Comment (Prearo) [inline src/main/java/Duke.java]: Try to avoid wildcard imports and try to add some header comments for your classes and methods.

File: src/main/java/Events.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+public class Events extends Task{
```
Comment (Prearo) [inline src/main/java/Events.java]: Does this represent one event or multiple events?

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -1,10 +1,143 @@
...
+
+            }
+
+            else {
+                taskArray.add(new Task(str));
+                counter++;
+                System.out.println("added: " + str);
+                str = sc.nextLine();
+            }
+
```
Comment (Prearo) [inline src/main/java/Duke.java]: Some whitespace is good, but try to avoid too many empty lines. I find it hard to follow.

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -1,10 +1,143 @@
...
+            } else {
+                return;
+            }
+        }
+    }
+
+    public void checkInput(String str) throws DukeException {
+        if (str.equals("")) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
```
Comment (Prearo) [inline src/main/java/Duke.java]: Try to stick to K&R style bracketing - else if should be on the same line as the closing } of the previous if clause.
Moreover, since each result is a thrown exception, there is no need for 'else'.
Also, what is 'blah' here?

File: src/main/java/Duke.java [hunk 6]
```diff
@@ -1,10 +1,143 @@
...
+            }
+
+            else if (str.equals("mark")) {
+                int num = sc.nextInt();
+                Task currTask = taskArray.get(num - 1);
+                currTask.mark();
+                System.out.println("Nice! I've marked this task as done:\n" +
+                        "[" + currTask.getStatusIcon() + "] " + currTask.description);
+                str = sc.nextLine();
+                str = sc.nextLine();
```
Comment (Prearo) [inline src/main/java/Duke.java]: Is there a reason for the double nextLine()? Is it due to the input format?

File: src/main/java/Duke.java [hunk 7]
```diff
@@ -1,10 +1,143 @@
...
+            counter++;
+            str = sc.nextLine();
+        }
+
+
+
+
+    }
+
+    public void returnArrayStrings(ArrayList<Task> tskArray){
```
Comment (Prearo) [inline src/main/java/Duke.java]: I think naming the parameter taskArray here is at least clearer.

File: src/main/java/Duke.java [hunk 8]
```diff
@@ -1,10 +1,143 @@
...
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello from\n" + logo);*/
     }
+
+    public void chatDuke() throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        ArrayList<Task> taskArray = new ArrayList<>();
+        String str = sc.nextLine();
+        int counter = 0;
```
Comment (Prearo) [inline src/main/java/Duke.java]: Consider using ArrayList.size() to replace an explicit counter variable to get the last task in the list.

File: src/main/java/Duke.java [hunk 9]
```diff
@@ -1,10 +1,143 @@
...
+
+
+
+
+    }
+
+    public void returnArrayStrings(ArrayList<Task> tskArray){
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tskArray.size(); i++){
+            if (tskArray.get(i) != null) {
```
Comment (sikai00) [inline src/main/java/Duke.java]: Nitpick here, but might be good to avoid using ```tsk.Array.get(i)``` twice. Instead, assign its return value to a variable.

File: src/main/java/Duke.java [hunk 10]
```diff
@@ -1,10 +1,143 @@
...
+                System.out.println(Integer.toString(i + 1) + "." + tskArray.get(i));
+            } else {
+                return;
```
Comment (sikai00) [inline src/main/java/Duke.java]: Given that the for loop runs for tskArray.size(), is it really necessary to try and break with ```return;```?

File: src/main/java/Duke.java [hunk 11]
```diff
@@ -1,10 +1,143 @@
...
+                        taskArray.get(counter));
+                counter++;
+                System.out.println("Now you have " + counter + " tasks in the list.");
+                str = sc.nextLine();
+            }
+            else if (str.equals("deadline")) {
+                String input = sc.nextLine();
+                checkInput(input);
+                String by = sc.nextLine();
+                checkInput(by);
```
Comment (sikai00) [inline src/main/java/Duke.java]: ```checkInput```'s exception message does not seem to match the argument being used (not a todo?). More importantly, there seems like better and cleaner ways to check the validity of inputs.

Comment (sikai00) [reply src/main/java/Duke.java]: It also seems like there is no indicator for the user to know what to enter at which point, since your arguments are all separated by a newline. Maybe some feedback would be good for UX?

**Review by sikai00 — COMMENTED** [review]
Most of the comments I had were already pointed out @Prearo. There's a lot of room for improvement in my opinion, in terms of cleaner code. More effort and work should also be put in for the styling of code.
